packages/firestore/lib/FirestoreQueryModifiers.js

```diff
@@ -285,15 +285,21 @@ export default class FirestoreQueryModifiers {
       for (let k = 0; k < this._orders.length; k++) {
         const order = this._orders[k];
         const orderFieldPath = order.fieldPath;
-        // Any where() fieldPath parameter cannot match any orderBy() parameter
         if (filter.operator === OPERATORS['==']) {
+          // Any where() fieldPath parameter cannot match any orderBy() parameter when '==' operand is invoked
           if (filterFieldPath === orderFieldPath) {
             throw new Error(
               `Invalid query. Query.orderBy() parameter: ${orderFieldPath} cannot be the same as your Query.where() fieldPath parameter: ${filterFieldPath}`,
             );
           }
         }
 
+        if (filterFieldPath === '__name__' && orderFieldPath !== '__name__') {
```

Comment:
Could you directly reference the `documentId` method on the FirestoreFieldPath class here instead of the string name directly?

Reply:
sure, good idea

Reply:
@Ehesp not sure that is possible. That method is available to the firestore web SDK which we don't use. I would suggest we make it a constant  and replace it in other places we have it hardcoded.